### PR TITLE
Initial setup

### PR DESCRIPTION
**Pull request by cardel**
> Autenticación en GitHub
> Consumiendo Métodos GET
> Consumiendo Métodos PUT
> Consumiendo métodos POST y PATCH
> Consumiendo un DELETE y un recurso inexistente
>

### CODE REVIEW

File: test/GithubApi.Repositories.test.js [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    });
+
+    it('Verify user data', async () => {
+      expect(user.name).to.equal('Alejandro Perdomo');
+      expect(user.company).to.equal('Perficient Latam');
+      expect(user.location).to.equal('Colombia');
+    });
+
+    describe('Get repository', async () => {
+      let repos;
```
Comment (holgiosalos) [inline test/GithubApi.Repositories.test.js]: as this variable is only used in the response, you can declare it within that scope

File: test/GithubApi.Repositories.test.js [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+          files = response.data;
+          readmeFile = files.find((f) => f.name === 'README.md');
+        });
+        it('Verify existence README.md file', async () => {
+          assert.exists(readmeFile);
+        });
+        it('Verify content README.md file', async () => {
+          expect(md5(readmeFile)).to.equal(md5hashReadMe);
+        });
+      });
```
Comment (holgiosalos) [inline test/GithubApi.Repositories.test.js]: could you create a describe for the verifications related to the readme file?

File: test/GithubApi.Repositories.test.js [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+        const md5hashReadMe = '3449c9e5e332f1dbb81505cd739fbf3f';
+        before(async () => {
+          const response = await axios.get(`${repo.url}/contents`);
+          files = response.data;
+          readmeFile = files.find((f) => f.name === 'README.md');
+        });
+        it('Verify existence README.md file', async () => {
+          assert.exists(readmeFile);
+        });
+        it('Verify content README.md file', async () => {
```
Comment (holgiosalos) [inline test/GithubApi.Repositories.test.js]: add one more test, in which you should use the containSubset to verify the name, path and sha of the readme file.

File: test/GithubApi.Issue.test.js [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+  };
+  before(async () => {
+    const response = await axios.get(`${url}/user`, authHeaders);
+    user = response.data;
+  });
+  it('The user has public repositories ', async () => {
+    expect(user.public_repos).to.be.above(0);
+  });
+  describe('Get all repos', async () => {
+    let repo;
```
Comment (holgiosalos) [inline test/GithubApi.Issue.test.js]: let's rename this variable to increase the readability: `cursosRepoData`

File: test/GithubApi.Issue.test.js [hunk 5]
```diff
@@ -0,0 +1,68 @@
+const axios = require('axios');
+const { expect } = require('chai');
+
+describe('Get an authenticated user', async () => {
+  let user;
+  const url = 'https://api.github.com';
+  const repoObj = 'Cursos';
```
Comment (holgiosalos) [inline test/GithubApi.Issue.test.js]: repoObjName

File: test/GithubApi.Issue.test.js [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+      const response = await axios.get(user.repos_url, authHeaders);
+      repo = response.data.find((r) => r.name === repoObj);
+    });
+
+    it('We have de repository', async () => {
+      expect(repo).to.not.equal(undefined);
+    });
+
+    describe('Create a new issue', async () => {
+      const issue = { title: 'A beautiful issue' };
```
Comment (holgiosalos) [inline test/GithubApi.Issue.test.js]: let's rename it as `issueCreationBody`

File: test/GithubApi.Issue.test.js [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+      const updatedIssue = { body: 'A nice body' };
```
Comment (holgiosalos) [inline test/GithubApi.Issue.test.js]: let's rename it as `issueUpdateBody`

File: test/GithubApi.Issue.test.js [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+      let iss;
```
Comment (holgiosalos) [inline test/GithubApi.Issue.test.js]: let's rename it as `createdIssueData`

File: test/GithubApi.Repositories.test.js (hunk 1 above)
Comment (cardel) [reply test/GithubApi.Repositories.test.js]: Fixed
`
const repos = response.data;
`

File: test/GithubApi.Repositories.test.js (hunk 2 above)
Comment (cardel) [reply test/GithubApi.Repositories.test.js]: Fixed
`
describe('Verifications related to the README.md file', async () => {
`

File: test/GithubApi.Repositories.test.js (hunk 3 above)
Comment (cardel) [reply test/GithubApi.Repositories.test.js]: Fixed
`
          const contentReadMe = {
            name: 'README.md',
            path: 'README.md',
            sha: '360eee6c223cee31e2a59632a2bb9e710a52cdc0'
          };

expect(readmeFile).to.containSubset(contentReadMe);
`

File: test/GithubApi.Issue.test.js (hunk 4 above)
Comment (cardel) [reply test/GithubApi.Issue.test.js]: Fixed

File: test/GithubApi.Issue.test.js (hunk 5 above)
Comment (cardel) [reply test/GithubApi.Issue.test.js]: Fixed

File: test/GithubApi.Issue.test.js (hunk 6 above)
Comment (cardel) [reply test/GithubApi.Issue.test.js]: Fixed

File: test/GithubApi.Issue.test.js (hunk 7 above)
Comment (cardel) [reply test/GithubApi.Issue.test.js]: Fixed

File: test/GithubApi.Issue.test.js (hunk 8 above)
Comment (cardel) [reply test/GithubApi.Issue.test.js]: Fixed

**Comment by cardel** [comment]
Thanks four your review, I fixed the code according your comments.

**Review by holgiosalos — APPROVED** [review]

**Pull request merged**